### PR TITLE
sort: show large numbers instead of `inf`

### PR DESCRIPTION
**Pull request by NaveenKarippai**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.
- [ ] New Instant Answer
  - [ ] Cheat Sheets: **`New {Cheat Sheet Name} Cheat Sheet`**
  - [ ] Other: **`New {IA Name} Instant Answer`**
- [x] Improvement
  - [x] Bug fix: **`sort: Fix {#3145} show large numbers`**
  - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
- [ ] Non–Instant Answer
  - [ ] Other (Role, Template, Test, Documentation, etc.): **`{GoodieRole/Templates/Tests/Docs}: {Short Description}`**
###### Description of changes

The sort_goodie showed `inf` for large numbers such as 1e1000.
This could be fixed by tweaking normalise function on module.

Provide an overview of the changes this pull request introduces.
- shows large numbers instead of `inf` unknown
- test suite adapted
###### Which issues (if any) does this fix?

See also: #3145

Fixes #NNNN - how specifically does it fix the issue?
###### People to notify (@GuiltyDolphin)

---

Instant Answer Page: https://duck.co/ia/view/sort

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @koosha--

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: t/Sort.t [hunk 1]
```diff
@@ -24,14 +24,14 @@ sub build_test {
...
+    'sort -1, +4, -3, 5.7'   => build_test('-1, +4, -3, 5.7', 'ascendingly', '-3, -1, +4, 5.7'),
+    'sort [-1, +4, -3, 5.7]' => build_test('-1, +4, -3, 5.7', 'ascendingly', '-3, -1, +4, 5.7'),
+    'sort (-1, +4, -3, 5.7)' => build_test('-1, +4, -3, 5.7', 'ascendingly', '-3, -1, +4, 5.7'),
+    'sort desc -4.4 .5 1 66 15 -55' => build_test('-4.4, .5, 1, 66, 15, -55', 'descendingly', '66, 15, 1, .5, -4.4, -55'),
+    'sort desc -4.4 .5 1 66 2e-3 15 -55' => build_test('-4.4, .5, 1, 66, 2e-3, 15, -55', 'descendingly', '66, 15, 1, .5, 2e-3, -4.4, -55'),
     'sort -3 -10 56 10' => build_test('-3, -10, 56, 10','ascendingly', '-10, -3, 10, 56'),
-    'sort descending 10, -1, +5.3, -95, 1' => build_test('10, -1, 5.3, -95, 1', 'descendingly', '10, 5.3, 1, -1, -95'),
-    'sort descending 10, -1, +5.3, -95, 1, 1e2' => build_test('10, -1, 5.3, -95, 1, 100', 'descendingly', '100, 10, 5.3, 1, -1, -95'),
+    'sort descending 10, -1, +5.3, -95, 1' => build_test('10, -1, +5.3, -95, 1', 'descendingly', '10, +5.3, 1, -1, -95'),
+    'sort descending 10, -1, +5.3, -95, 1, 1e2' => build_test('10, -1, +5.3, -95, 1, 1e2', 'descendingly', '1e2, 10, +5.3, 1, -1, -95'),
```
Comment (GuiltyDolphin) [inline t/Sort.t]: We should add tests for `sort 1e700, 1e500, 1e-700` etc.

File: t/Sort.t [hunk 2]
```diff
@@ -24,14 +24,14 @@ sub build_test {
...
 ddg_goodie_test(
     [qw( DDG::Goodie::Sort)],
-    'sort -1, +4, -3, 5.7'   => build_test('-1, 4, -3, 5.7', 'ascendingly', '-3, -1, 4, 5.7'),
-    'sort [-1, +4, -3, 5.7]' => build_test('-1, 4, -3, 5.7', 'ascendingly', '-3, -1, 4, 5.7'),
-    'sort (-1, +4, -3, 5.7)' => build_test('-1, 4, -3, 5.7', 'ascendingly', '-3, -1, 4, 5.7'),
-    'sort desc -4.4 .5 1 66 15 -55' => build_test('-4.4, 0.5, 1, 66, 15, -55', 'descendingly', '66, 15, 1, 0.5, -4.4, -55'),
-    'sort desc -4.4 .5 1 66 2e-3 15 -55' => build_test('-4.4, 0.5, 1, 66, 0.002, 15, -55', 'descendingly', '66, 15, 1, 0.5, 0.002, -4.4, -55'),
+    'sort -1, +4, -3, 5.7'   => build_test('-1, +4, -3, 5.7', 'ascendingly', '-3, -1, +4, 5.7'),
+    'sort [-1, +4, -3, 5.7]' => build_test('-1, +4, -3, 5.7', 'ascendingly', '-3, -1, +4, 5.7'),
+    'sort (-1, +4, -3, 5.7)' => build_test('-1, +4, -3, 5.7', 'ascendingly', '-3, -1, +4, 5.7'),
```
Comment (GuiltyDolphin) [inline t/Sort.t]: We should be using the NumberStyle role to format these numbers.

Comment (NaveenKarippai) [inline t/Sort.t]: @GuiltyDolphin could you please be more specific about NumberStyle role. Do you mean signed numbers?

Comment (GuiltyDolphin) [inline t/Sort.t]: @NaveenKarippai I mean we should be using the NumberStyle role to *parse* and *format* the numbers, not doing custom formatting (which is also inconsistent).

Comment (NaveenKarippai) [inline t/Sort.t]: @GuiltyDolphin hm, would it be possible for you to throw a quick example or helpful docs? Thanks!

Comment (GuiltyDolphin) [inline t/Sort.t]: @NaveenKarippai There's some [docs](http://docs.duckduckhack.com/backend-reference/goodie-helpers.html#numbers) for NumberStyle (might be worth looking at the source and other Goodies too).

**Comment by moollaza** [comment]
@NaveenKarippai thanks for this -- looks like despite that last commit you ended up with a conflict 😞  -- care to update it again so I can get this onto beta?

**Comment by NaveenKarippai** [comment]
@moollaza yup, sure. I can update PR and try to resolve the conflict.

File: lib/DDG/Goodie/Sort.pm [hunk 3]
```diff
@@ -33,11 +33,12 @@ handle remainder => sub {
...
 
     my @numbers = split /[\s,;]+/, $input;
-
+    
+    return unless @numbers > 1;
     return unless all { looks_like_number($_) } @numbers;
 
     my $count = 0;
-    @numbers = map { 0 + $_ } grep { ++$count <= MAX_LIST_SIZE } @numbers;    # Normalize and limit list size.
+    @numbers = map { $_ } grep { ++$count <= MAX_LIST_SIZE } @numbers; # Normalize and limit list size.
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/Sort.pm]: Just `grep { ++$count <= MAX_LIST_SIZE } @numbers`.

Comment (NaveenKarippai) [inline lib/DDG/Goodie/Sort.pm]: @GuiltyDolphin true! The new changeset should fix this issue.

File: t/Sort.t (hunk 1 above)
Comment (NaveenKarippai) [inline t/Sort.t]: done!

**Comment by NaveenKarippai** [comment]
@moollaza @mintsoft @zekiel could you please review the changes. Thanks!

File: lib/DDG/Goodie/Sort.pm [hunk 4]
```diff
@@ -33,12 +33,12 @@ handle remainder => sub {
     }
 
     my @numbers = split /[\s,;]+/, $input;
-
+
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/Sort.pm]: Remove trailing whitespace.

File: lib/DDG/Goodie/Sort.pm [hunk 5]
```diff
@@ -33,12 +33,12 @@ handle remainder => sub {
     }
 
     my @numbers = split /[\s,;]+/, $input;
-
+    
     return unless @numbers > 1;
+    return if @numbers > MAX_LIST_SIZE;
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/Sort.pm]: `return unless @numbers > 1 && @numbers < MAX_LIST_SIZE`

Comment (NaveenKarippai) [inline lib/DDG/Goodie/Sort.pm]: @GuiltyDolphin Thanks for the review! New changeset pushed :8ball:

File: t/Sort.t [hunk 6]
```diff
@@ -24,19 +24,20 @@ sub build_test {
...
-    'sort' => undef,
+    'sort -1, +4, -3, 5.7'                      => build_test('-1, +4, -3, 5.7', 'ascendingly', '-3, -1, +4, 5.7'),
+    'sort [-1, +4, -3, 5.7]'                    => build_test('-1, +4, -3, 5.7', 'ascendingly', '-3, -1, +4, 5.7'),
+    'sort (-1, +4, -3, 5.7)'                    => build_test('-1, +4, -3, 5.7', 'ascendingly', '-3, -1, +4, 5.7'),
+    'sort desc -4.4 .5 1 66 15 -55'             => build_test('-4.4, .5, 1, 66, 15, -55', 'descendingly', '66, 15, 1, .5, -4.4, -55'),
+    'sort desc -4.4 .5 1 66 2e-3 15 -55'        => build_test('-4.4, .5, 1, 66, 2e-3, 15, -55', 'descendingly', '66, 15, 1, .5, 2e-3, -4.4, -55'),
+    'sort -3 -10 56 10'                         => build_test('-3, -10, 56, 10','ascendingly', '-10, -3, 10, 56'),
+    'sort descending 10, -1, +5.3, -95, 1'      => build_test('10, -1, +5.3, -95, 1', 'descendingly', '10, +5.3, 1, -1, -95'),
+    'sort descending 10, -1, +5.3, -95, 1, 1e2' => build_test('10, -1, +5.3, -95, 1, 1e2', 'descendingly', '1e2, 10, +5.3, 1, -1, -95'),
+    'sort 1e700, 1e500, 1e-700'                 => build_test('1e700, 1e500, 1e-700', 'ascendingly','1e-700, 1e700, 1e500'),
```
Comment (GuiltyDolphin) [inline t/Sort.t]: Uhh, this is incorrect; `1e700` should be *after* `1e500`.

Comment (moollaza) [inline t/Sort.t]: @GuiltyDolphin Agreed.

@NaveenKarippai the order should be `1e700 (0), 1e500 (inf), 1e700(bigger inf)`

**Comment by moollaza** [comment]
Deploying to Beta...

@NaveenKarippai I think once those tests are corrected this should be good to go?

File: t/Sort.t (hunk 6 above)
Comment (NaveenKarippai) [inline t/Sort.t]: @moollaza true!

@GuiltyDolphin Surprisingly, the goodie returns the (sorted) result in this specific order. what would you say?

Comment (GuiltyDolphin) [inline t/Sort.t]: @NaveenKarippai I'd say fix it :)

You should be able to use `Math::Big(Float|Rat|Int)` for this kind of thing, it seems to cope pretty well.

**Comment by moollaza** [comment]
@NaveenKarippai this is now live on Beta. Once that sorting is corrected we can merge this one 👍

**Comment by mintsoft** [comment]
@NaveenKarippai Just following up; this is so close to being ready!

**Comment by mintsoft** [comment]
I'm closing this PR for now, happy to reopen if wanted

**Pull request closed without merging**